packages/runtime-core/src/apiDefineComponent.ts

```diff
@@ -198,6 +222,11 @@ export function defineComponent<
 ): DefineComponent<PropsOptions, RawBindings, D, C, M, Mixin, Extends, E, EE>
 
 // implementation, close to no-op
-export function defineComponent(options: unknown) {
-  return isFunction(options) ? { setup: options, name: options.name } : options
+export function defineComponent(
+  options: unknown,
+  extraOptions?: ComponentOptions
+) {
+  return isFunction(options)
+    ? extend({}, extraOptions, { setup: options, name: options.name })
```

Comment:
It would not be set correct name, if `options` is anonymous function and set name in `extraOptions`

e.g

```ts
const Comp = defineComponent(() => {
  // ...
}, {
  name: 'Foo',
})
```

Reply:
Thanks for the catch. Addressed in b2be75ba